### ISSUE TITLE
1.7.0.Alpha7 does not call JUnit lifecycle hooks?

### ISSUE THREAD

**Issue by hantsy** [arquillian/arquillian-core#309]
I maintained a [jakartaee9 starter template](https://github.com/hantsy/jakartaee9-starter-boilerplate) which is provided to kickstart a Jakarta EE 9 project.

I am not sure what is changed in the latest 1.7.0.Alpha7 in today's PR created by Renovate.

From 1.7.0.Alpha6->1.7.0.Alpha7, the following simple test failed due to a NullPointerException. 

```java
@ExtendWith(ArquillianExtension.class)
public class GreetingResourceTest {
    private final static Logger LOGGER = Logger.getLogger(GreetingResourceTest.class.getName());
    
    @Deployment(testable = false)
    public static WebArchive createDeployment() {
        return ShrinkWrap.create(WebArchive.class)
                .addClass(GreetingMessage.class)
                .addClass(GreetingService.class)
                .addClasses(GreetingResource.class, JaxrsActivator.class)
                // Enable CDI (Optional since Java EE 7.0)
                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
    }
    
    @ArquillianResource
    private URL base;
    
    private Client client;
    
    @BeforeEach
    public void setup() {
        LOGGER.info("call BeforeEach");
        this.client = ClientBuilder.newClient();
        //removed the Jackson json provider registry, due to OpenLiberty 21.0.0.1 switched to use Resteasy.
    }
    
    @AfterEach
    public void teardown() {
        LOGGER.info("call AfterEach");
        if (this.client != null) {
            this.client.close();
        }
    }
    
    @Test
    @DisplayName("Given a name:`JakartaEE` should return `Say Hello to JakartaEE`")
    public void should_create_greeting() throws MalformedURLException {
        LOGGER.log(Level.INFO, " client: {0}, baseURL: {1}", new Object[]{client, base});
        final WebTarget greetingTarget = this.client.target(new URL(this.base, "api/greeting/JakartaEE").toExternalForm());
        try (final Response greetingGetResponse = greetingTarget.request()
                .accept(MediaType.APPLICATION_JSON)
                .get()) {
            assertThat(greetingGetResponse.getStatus()).isEqualTo(200);
            assertThat(greetingGetResponse.readEntity(GreetingMessage.class).getMessage()).startsWith("Say Hello to JakartaEE");
        }
    }
}
```
The client is **null** there in the test method `should_create_greeting`.

Check the build result from Github actions(Ubuntu, Java 8).

https://github.com/hantsy/jakartaee9-starter-boilerplate/pull/99

**Comment by lprimak** [comment]
I can confirm this. Relates to #288 and #301
I am working on a reproducer for it

**Comment by hantsy** [comment]
It seems it is occurred only tests run in client mode.

**Comment by lprimak** [comment]
Ok, @kifj This is critical.
To summarise:
- No lifecycle methods are called in client mode ( as in example above)
- `@Before/AfterAll` methods are called on the container, as they are called on the client in JUnit 4 (that behavior should be kept)
- `@Before/AfterAll` methods are called for each test, not once-per-class

**Comment by lprimak** [comment]
@kifj Was an easy fix, so I did it :)

**Comment by kifj** [comment]
Thank you!

**Comment by hantsy** [comment]
I think this is a blocking issue. If possible, please publish a quick fix version to replace 1.7.0.Alpha7.

**Comment by bartoszmajsak** [comment]
> I think this is a blocking issue. If possible, please publish a quick fix version to replace 1.7.0.Alpha7.

Coming soon.
